File: SupportDesk.API/Models/TicketHistory.cs
namespace SupportDesk.API.Models;

public class TicketHistory
{
    public int Id { get; set; }

    public int TicketId { get; set; }
    public Ticket? Ticket { get; set; }

    public int UserId { get; set; }
    public User? User { get; set; }

    public string Action { get; set; } = string.Empty;

    public string? OldValue { get; set; }

    public string? NewValue { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public string? FieldName { get; set; }
}

File: SupportDesk.Web/src/types/ticketHistory.ts
export interface TicketHistoryItem {
  id: number;
  ticketId: number;
  userId: number;
  action: string;
  oldValue: string | null;
  newValue: string | null;
  createdAt: string;
  user: {
    id: number;
    name: string;
    email: string;
    role: string;
    isActive: boolean;
  } | null;
  fieldName: string | null;
}

export interface PagedTicketHistoryResponse {
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
  items: TicketHistoryItem[];
}